### DISCUSSION TITLE
PLEASE FIX CONFIG_LIST

### DISCUSSION THREAD

**Discussion by squatchydev9000** [microsoft/autogen#1163] [General]
I have surrendered (for now) and will only be trying OAI models... no open-source drama for a while...

HOWEVER - I CANNOT GET the OAI_CONFIG_LIST nonsesne to work... in any of the 4 formats you provide...

Current headache: "[autogen.oai.client: 01-06 14:48:06] {80} WARNING - openai client was provided with an empty config_list, which may not be intended."

The Config_List HAS to be your biggest issue/complaint with autogen....

**Comment by rickyloynd-microsoft** [comment]
Can you paste your OAI_CONFIG_LIST file here, after removing the api key strings?

**Reply by squatchydev9000** [reply]
[
    {
        "model": "gpt-4",
        "api_key": "specific-gpt-4-key"  
    },
    {
        "model": "gpt-3.5-turbo",
        "api_key": "specific-gpt-3.5-turbo-key" 
    },
    {
        "model": "tts-1-hd",
        "api_key": "specific-tts-key"
    },
    {
        "model":"gpt-4-vision-preview",
        "api_key": "specific-gpt-4-vision-key"
    },
    {   "model":"dall-e-3",
        "api_key":"specific-dalle-3-key" 
    }
]

**Reply by squatchydev9000** [reply]
its named OAI_CONFIG_LIST and it is within the same folder as the file I am trying to run... here is that code:

import autogen
from autogen import AssistantAgent, UserProxyAgent

cheap_config_list = autogen.config_list_from_json(
    env_or_file="OAI_CONFIG_LIST",
    filter_dict={
        "model": {
            "gpt-3.5-turbo",
        }
    }
)

assistant = AssistantAgent("assistant", llm_config={"config_list": cheap_config_list})
user_proxy = UserProxyAgent("user_proxy", code_execution_config={"work_dir": "coding"})
user_proxy.initiate_chat(assistant, message="Plot a chart of NVDA and TESLA stock price change YTD.")

**Reply by squatchydev9000** [reply]
using python 3.11.6 inside a poetry venv....

**Reply by squatchydev9000** [reply]
* if it helps - when trying the from dot_env option I get "WARNING:root:API key not found or empty for model gpt-3.5-turbo. Please ensure path to .env file is correct.
ERROR:root:No configurations loaded."

**Reply by sonichi** [reply]
Do you have an env variable named "OAI_CONFIG_LIST"? The env var has higher priority than the file.

**Reply by squatchydev9000** [reply]
.env is also located within the same folder as the file I am attempting to run...

**Reply by sonichi** [reply]
`filter_dict={"model": ["gpt-3.5-turbo"]}`
Note the []

**Reply by squatchydev9000** [reply]
same result... autogen cannot see the contents of the config list for some reason... 
thanks for trying guys.

**Reply by squatchydev9000** [reply]
***UPDATE*** - got things working error free - there was a poetry environment issue...

Cheers.
